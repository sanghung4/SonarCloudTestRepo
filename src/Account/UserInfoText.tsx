import React from 'react';

import {
  Grid,
  Typography,
  useScreenSize
} from '@dialexa/reece-component-library';

interface Props {
  fieldName: String;
  fieldValue: String;
  color: any;
  testId: string;
}

function UserInfoText(props: Props) {
  /**
   * Custom Hooks
   */
  const { isSmallScreen } = useScreenSize();

  return (
    <Grid
      container
      direction="row"
      spacing={2}
      justifyContent="space-between"
      wrap="nowrap"
      sx={{
        width: 1,
        px: isSmallScreen ? 3 : 5,
        py: 2.5
      }}
    >
      <Grid item>
        <Typography noWrap variant={isSmallScreen ? 'body2' : 'body1'}>
          {props.fieldName}
        </Typography>
      </Grid>
      <Grid item zeroMinWidth>
        <Typography
          noWrap
          align="right"
          variant={isSmallScreen ? 'body2' : 'body1'}
          color={props.color}
          sx={{ fontWeight: 700 }}
          data-testid={props.testId}
        >
          {props.fieldValue}
        </Typography>
      </Grid>
    </Grid>
  );
}

export default UserInfoText;
