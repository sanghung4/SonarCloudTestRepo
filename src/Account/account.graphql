query user($userId: String) {
  user(userId: $userId) {
    id
    email
    firstName
    lastName
    phoneNumber
    role {
      id
      name
      description
    }
    phoneType
  }
}

query UserAccounts($userId: String) {
  userAccounts(userId: $userId) {
    id
    name
    erpAccountId
    erpSystemName
    shipTos {
      id
      name
      erpAccountId
      erpSystemName
      divisionEnum
      branchId
      branchAddress
    }
    erpSystemName
    divisionEnum
    branchId
    branchAddress
  }
}

mutation updateUserPassword(
  $updateUserPasswordInput: UpdateUserPasswordInput!
) {
  updateUserPassword(updateUserPasswordInput: $updateUserPasswordInput)
}
