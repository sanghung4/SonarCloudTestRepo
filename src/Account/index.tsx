import React, { useContext, useEffect, useMemo, useState } from 'react';

import {
  Box,
  Breadcrumbs,
  Button,
  Container,
  Grid,
  Hidden,
  Link,
  Typography,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

import { AuthContext } from 'AuthProvider';
import Loader from 'common/Loader';
import {
  ErpAccount,
  useAccountDetailsQuery,
  useUserQuery
} from 'generated/graphql';
import { ChevronLeftIcon, ChevronRightIcon } from 'icons';
import ViewAccount from 'Account/ViewAccount';
import ViewCompanyInfo from 'Account/ViewCompanyInfo';
import EditAccount from 'Account/EditAccount';
import { useDomainInfo } from 'hooks/useDomainInfo';
import useDocumentTitle from 'hooks/useDocumentTitle';
import { useSelectedAccountsContext } from 'providers/SelectedAccountsProvider';

function User() {
  /**
   * Custom Hooks
   */
  const history = useHistory();
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();
  const { brand } = useDomainInfo();
  useDocumentTitle(t('common.accountManagement'));

  /**
   * Context
   */
  const { profile } = useContext(AuthContext);
  const { selectedAccounts } = useSelectedAccountsContext();

  /**
   * Data fetching
   */
  const { data: userQuery, loading: userLoading } = useUserQuery({
    variables: {
      userId: profile?.userId
    }
  });
  const { data: accountQuery, loading: accountLoading } =
    useAccountDetailsQuery({
      variables: {
        accountId: selectedAccounts?.billTo?.id ?? '',
        brand
      }
    });

  /**
   * State
   */
  const [isEditing, setIsEditing] = useState(false);
  const [userInfo, setUserInfo] = useState({
    firstName: '',
    lastName: '',
    phoneNumber: '',
    phoneTypeId: '',
    email: '',
    role: '',
    id: ''
  });

  /**
   * Callbacks
   */
  const handleEditClicked = () => setIsEditing(true);
  const handleFinishEditing = (values?: any) => {
    values && setUserInfo({ ...userInfo, ...values });
    setIsEditing(false);
  };

  const companyInfo = useMemo(accountMemo, [
    accountQuery,
    selectedAccounts.erpSystemName
  ]);

  /**
   * Effects
   */
  useEffect(() => {
    if (userQuery?.user) {
      setUserInfo({
        firstName: userQuery.user.firstName!,
        lastName: userQuery.user.lastName!,
        phoneNumber: userQuery.user.phoneNumber!,
        phoneTypeId: userQuery.user.phoneType!,
        email: userQuery.user.email!,
        role: userQuery.user.role?.name!,
        id: userQuery.user.id!
      });
    }
  }, [userQuery]);

  return userLoading ||
    accountLoading ||
    !userQuery?.user ||
    !accountQuery?.account ||
    !accountQuery?.account[0] ? (
    <Loader />
  ) : (
    <Box mb={5}>
      <Container maxWidth="lg" disableGutters={isSmallScreen}>
        <Grid container item spacing={4}>
          <Hidden mdUp>
            <Grid item xs={12}>
              <Button
                sx={{ mt: 1 }}
                startIcon={<ChevronLeftIcon />}
                variant="inline"
                onClick={onBackToHome}
              >
                {t('common.backToHome')}
              </Button>
            </Grid>
          </Hidden>
          <Hidden mdDown>
            <Grid item xs={12}>
              <Breadcrumbs
                aria-label="breadcrumb"
                separator={<ChevronRightIcon />}
              >
                <Link color="primary" href="/">
                  {t('common.home')}
                </Link>
                <Typography color="inherit" variant="caption">
                  {t('common.accountInfo')}
                </Typography>
              </Breadcrumbs>
            </Grid>
          </Hidden>
          <Grid item xs={12}>
            <Box mb={2} pl={isSmallScreen ? 3 : 0}>
              <Typography color="textPrimary" variant="h4">
                {`${userInfo.firstName} ${userInfo.lastName}`}
              </Typography>
            </Box>
          </Grid>
          <Grid item xs={12} md={6}>
            {isEditing ? (
              <EditAccount user={userInfo} onFinished={handleFinishEditing} />
            ) : (
              <ViewAccount user={userInfo} onEditClicked={handleEditClicked} />
            )}
          </Grid>
          <Grid item xs={12} md={6}>
            <ViewCompanyInfo company={companyInfo} />
          </Grid>
        </Grid>
      </Container>
    </Box>
  );

  function onBackToHome() {
    history.push('/');
  }

  function accountMemo(): ErpAccount {
    const erpSystemName = selectedAccounts.erpSystemName;
    if (accountQuery?.account && erpSystemName) {
      const erpAccount = accountQuery?.account?.filter(
        (i) => i?.erpName === erpSystemName
      )[0];
      return erpAccount as ErpAccount;
    }
    return [] as ErpAccount;
  }
}

export default User;
