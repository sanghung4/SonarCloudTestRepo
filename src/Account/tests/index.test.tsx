import { OktaAuth } from '@okta/okta-auth-js';
import { waitFor } from '@testing-library/react';

import Account from 'Account';
import { ErpSystemEnum } from 'generated/graphql';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';
import { mockAccountDetails, mockUserwithMultiAccounts } from './mocks';

export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  oktaAuth: new OktaAuth({ issuer: 'http://test.com' }),
  profile: {
    userId: '11111111-1111-1111-1111-111111111111',
    permissions: [],
    isEmployee: false,
    isVerified: true
  }
};

describe('Account Tests', () => {
  it('should render the Account information page', async () => {
    const { container, getByTestId } = render(<Account />, {
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { erpAccountId: '' },
          shipTo: { erpAccountId: '' },
          shippingBranchId: ''
        }
      }
    });

    await waitFor(() =>
      expect(getByTestId('loader-component')).toBeInTheDocument()
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the Account information page with correct company info for Mincron User', async () => {
    setBreakpoint('desktop');
    const { container, getByTestId, queryByText } = render(<Account />, {
      authConfig,
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: {
            id: '22222222-2222-2222-2222-222222222222',
            erpAccountId: ''
          },
          shipTo: { erpAccountId: '' },
          shippingBranchId: '',
          erpSystemName: ErpSystemEnum.Mincron
        }
      },
      mocks: [mockUserwithMultiAccounts, mockAccountDetails]
    });

    await waitFor(() =>
      expect(getByTestId('editaccount-button')).toBeInTheDocument()
    );

    await waitFor(() => {
      expect(queryByText('HORIZON PLUMBING LTD')).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it('should render the Account information page with correct company info for Eclipse User', async () => {
    setBreakpoint('mobile');
    const { container, queryByText } = render(<Account />, {
      authConfig,
      mocks: [mockUserwithMultiAccounts, mockAccountDetails],
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: {
            id: '22222222-2222-2222-2222-222222222222',
            erpAccountId: ''
          },
          shipTo: { erpAccountId: '' },
          shippingBranchId: '',
          erpSystemName: ErpSystemEnum.Eclipse
        }
      }
    });

    await waitFor(() => {
      expect(queryByText('TEXAS MILITARY DEPARTMENT SHOP')).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
