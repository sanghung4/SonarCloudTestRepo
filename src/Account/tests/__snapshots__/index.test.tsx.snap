// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Tests should render the Account information page 1`] = `
<div>
  <div
    class="MuiBox-root css-1nen2z4"
    data-testid="loader-component"
  >
    <div
      class="MuiBox-root css-hyxlzm"
    >
      <span
        aria-valuenow="100"
        class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1wyqyef-MuiCircularProgress-root"
        role="progressbar"
        style="width: 80px; height: 80px; transform: rotate(-90deg);"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="19.5"
            stroke-width="5"
            style="stroke-dasharray: 122.522; stroke-dashoffset: 0.000px;"
          />
        </svg>
      </span>
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1gse1c1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 80px; height: 80px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="19.5"
            stroke-width="5"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`Account Tests should render the Account information page with correct company info for Eclipse User 1`] = `
<div>
  <div
    class="MuiBox-root css-10ib5jr"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-3ahet8-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-4 css-13twzmd-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1bbe40g-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
            >
              <svg>
                chevron-left.svg
              </svg>
            </span>
            Back to Home
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-7blwb6"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-ik5si6-MuiTypography-root"
            >
              Morsco EmployeeDev
            </h4>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1m6pi2w-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
            data-testid="viewaccount-component"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiCard-root css-1f2fbl-MuiPaper-root-MuiCard-root"
            >
              <div
                class="MuiBox-root css-hpgf8j"
              >
                <div
                  class="MuiCardHeader-root css-1bnehoc-MuiCardHeader-root"
                  data-testid="account-information-title"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-123mujq-MuiTypography-root"
                    >
                      Account Information
                    </span>
                  </div>
                  <div
                    class="MuiCardHeader-action css-nrv06q-MuiCardHeader-action"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-1unv2py-MuiButtonBase-root-MuiIconButton-root"
                      data-testid="editaccount-button"
                      tabindex="0"
                      type="button"
                    >
                      <svg>
                        edit.svg
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-1mkyrz5-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-15g5syf-MuiTypography-root"
                  >
                     Name: 
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-alignRight MuiTypography-noWrap css-1mia2g5-MuiTypography-root"
                    data-testid="name-field"
                  >
                    Morsco EmployeeDev
                  </p>
                </div>
              </div>
              <div
                class="MuiBox-root css-1bbcutn"
              >
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-1mkyrz5-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-15g5syf-MuiTypography-root"
                  >
                     Phone Number:
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-alignRight MuiTypography-noWrap css-1mia2g5-MuiTypography-root"
                    data-testid="phone-number-field"
                  >
                    (111) 111-1111
                  </p>
                </div>
              </div>
              <div
                class="MuiBox-root css-1bbcutn"
              >
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-1mkyrz5-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-15g5syf-MuiTypography-root"
                  >
                     Email:
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-alignRight MuiTypography-noWrap css-1mia2g5-MuiTypography-root"
                    data-testid="email-field"
                  >
                    john.doe@morsco.com
                  </p>
                </div>
              </div>
              <div
                class="MuiBox-root css-1bbcutn"
              >
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-1mkyrz5-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-15g5syf-MuiTypography-root"
                  >
                     Role: 
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-alignRight MuiTypography-noWrap css-1mia2g5-MuiTypography-root"
                    data-testid="role-field"
                  >
                    Morsco Admin
                  </p>
                </div>
              </div>
              <div
                class="MuiBox-root css-ff9sbb"
              >
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-wyox7y-MuiButtonBase-root-MuiButton-root"
                  data-testid="change-password-button"
                  tabindex="0"
                  type="button"
                >
                  Change password
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1m6pi2w-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiCard-root css-1f2fbl-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiBox-root css-fn7wg1"
            >
              <div
                class="MuiCardHeader-root css-1bnehoc-MuiCardHeader-root"
                data-testid="company-information-title"
              >
                <div
                  class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-123mujq-MuiTypography-root"
                  >
                    Company Information
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-1mkyrz5-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-15g5syf-MuiTypography-root"
                >
                  Company Name:
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-alignRight MuiTypography-noWrap css-2g732t-MuiTypography-root"
                  data-testid="company-name-field"
                >
                  TEXAS MILITARY DEPARTMENT SHOP
                </p>
              </div>
            </div>
            <div
              class="MuiBox-root css-1bbcutn"
            >
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-1mkyrz5-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-15g5syf-MuiTypography-root"
                >
                  Street Address:
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-alignRight MuiTypography-noWrap css-2g732t-MuiTypography-root"
                  data-testid="street-address-field"
                >
                  2200 W 35TH ST BLDG 11 2ND FLR AUSTIN, TX 78703-1222
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Account Tests should render the Account information page with correct company info for Mincron User 1`] = `
<div>
  <div
    class="MuiBox-root css-10ib5jr"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-cnms5y-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-4 css-13twzmd-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
        >
          <nav
            aria-label="breadcrumb"
            class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1yp6lty-MuiTypography-root-MuiBreadcrumbs-root"
          >
            <ol
              class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
            >
              <li
                class="MuiBreadcrumbs-li"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                aria-hidden="true"
                class="MuiBreadcrumbs-separator css-n1sla7-MuiBreadcrumbs-separator"
              >
                <svg>
                  chevron-right.svg
                </svg>
              </li>
              <li
                class="MuiBreadcrumbs-li"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-1gbxt6n-MuiTypography-root"
                >
                  Account Info
                </span>
              </li>
            </ol>
          </nav>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-10mpy3e"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-ik5si6-MuiTypography-root"
            >
              Morsco EmployeeDev
            </h4>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1m6pi2w-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
            data-testid="viewaccount-component"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-13gs6in-MuiPaper-root-MuiCard-root"
            >
              <div
                class="MuiBox-root css-hpgf8j"
              >
                <div
                  class="MuiCardHeader-root css-1bnehoc-MuiCardHeader-root"
                  data-testid="account-information-title"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h4 MuiCardHeader-title css-wa9bt7-MuiTypography-root"
                    >
                      Account Information
                    </span>
                  </div>
                  <div
                    class="MuiCardHeader-action css-nrv06q-MuiCardHeader-action"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-1unv2py-MuiButtonBase-root-MuiIconButton-root"
                      data-testid="editaccount-button"
                      tabindex="0"
                      type="button"
                    >
                      <svg>
                        edit.svg
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-t1bbgg-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1hdqiad-MuiTypography-root"
                  >
                     Name: 
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight MuiTypography-noWrap css-1arfpeq-MuiTypography-root"
                    data-testid="name-field"
                  >
                    Morsco EmployeeDev
                  </p>
                </div>
              </div>
              <div
                class="MuiBox-root css-1bbcutn"
              >
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-t1bbgg-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1hdqiad-MuiTypography-root"
                  >
                     Phone Number:
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight MuiTypography-noWrap css-1arfpeq-MuiTypography-root"
                    data-testid="phone-number-field"
                  >
                    (111) 111-1111
                  </p>
                </div>
              </div>
              <div
                class="MuiBox-root css-1bbcutn"
              >
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-t1bbgg-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1hdqiad-MuiTypography-root"
                  >
                     Email:
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight MuiTypography-noWrap css-1arfpeq-MuiTypography-root"
                    data-testid="email-field"
                  >
                    john.doe@morsco.com
                  </p>
                </div>
              </div>
              <div
                class="MuiBox-root css-1bbcutn"
              >
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-t1bbgg-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1hdqiad-MuiTypography-root"
                  >
                     Role: 
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight MuiTypography-noWrap css-1arfpeq-MuiTypography-root"
                    data-testid="role-field"
                  >
                    Morsco Admin
                  </p>
                </div>
              </div>
              <div
                class="MuiBox-root css-z6igcc"
              >
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-wyox7y-MuiButtonBase-root-MuiButton-root"
                  data-testid="change-password-button"
                  tabindex="0"
                  type="button"
                >
                  Change password
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1m6pi2w-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-13gs6in-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiBox-root css-fn7wg1"
            >
              <div
                class="MuiCardHeader-root css-1bnehoc-MuiCardHeader-root"
                data-testid="company-information-title"
              >
                <div
                  class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiTypography-h4 MuiCardHeader-title css-wa9bt7-MuiTypography-root"
                  >
                    Company Information
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-t1bbgg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1hdqiad-MuiTypography-root"
                >
                  Company Name:
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight MuiTypography-noWrap css-1d48ipw-MuiTypography-root"
                  data-testid="company-name-field"
                >
                  HORIZON PLUMBING LTD
                </p>
              </div>
            </div>
            <div
              class="MuiBox-root css-1bbcutn"
            >
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-t1bbgg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1hdqiad-MuiTypography-root"
                >
                  Street Address:
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-175wz9v-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight MuiTypography-noWrap css-1d48ipw-MuiTypography-root"
                  data-testid="street-address-field"
                >
                  2706 W PIONEER PKWY ARLINGTON, TX 76013-5906
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
